### DISCUSSION TITLE
useFormContext with yup and Material UI

### DISCUSSION THREAD

**Discussion by davidkipstar** [react-hook-form/react-hook-form#6199] [Q&A]
Hey everyone, 

I am refactoring my form logic and I want to make use of useFormContext to skip passing useState to child components.
My problem is that yup resolver is not working properly and is not reacting to input.
The goal is to create a yup scheme where i can easily exchange child components for less authorized users.

My yup schema

```tsx
const timeSchema = yup.object({
  date: yup.date().min(new Date()),
  start: yup.date(),
  end: yup.date().min(yup.ref("start"))
});
```
My parent component

```tsx
 <FormProvider {...methods}>
        <form>
          <TimeBlock start={new Date()} end={new Date()} date={new Date()} />
          <input type="submit" />
        </form>
        <DevTool control={methods.control} placement="top-right" />
      </FormProvider>
```
My child component look like this

```tsx
 <Controller
        render={({ field, fieldState, formState }) => {
          return (
            <KeyboardDatePicker
              {...field}
              clearable
              format="MM/dd/yyyy"
              label="Date"
              inputProps={{ min: 0, style: { textAlign: "center" } }}
              error={Boolean(fieldState.error)}
              helperText={fieldState.error?.message}
            />
          );
        }}
        name="date"
        control={control}
        defaultValue={date}
      />
```

[codesandbox](https://codesandbox.io/s/loving-ptolemy-9glmf?file=/src/timeBlock.tsx
)

https://codesandbox.io/s/loving-ptolemy-9glmf?file=/src/timeBlock.tsx

Thanks for tips, i think it's small change but I don't see it.

**Comment by jorisre** [comment]
I'm not sure to understand what you want to achieve, can you be more clear, please?

**Reply by davidkipstar** [reply]
Yes sure,

I did not understand the problem, I thought it was generally not working.
Before I was changing all values by datetimepicker Icon. When I did that no error messages were shown which gave me the impression something was wrong with my yup config.
But now when I  focus on the input representation after changing the value via datetimepicker I get the error messages I want to see.  Then in my dev tools touched is true and then I get the error message. 

So the problem is that changing the value by datetimepicker is not touching the form value which then doesn't get validated. 


I hope it is more clear now.
Thanks
